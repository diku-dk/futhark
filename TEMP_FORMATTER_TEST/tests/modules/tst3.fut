-- A functor that swaps types
-- ==
-- input {}
-- output { 3 }

module F (X : { type t type s }) : { type t = X.s type s = X.t } =
   { type t = X.s type s = X.t }

module A = { type t = f32 type s = i32 }
module B = F(A)
module C = F(B)

def main : i32 = 3
