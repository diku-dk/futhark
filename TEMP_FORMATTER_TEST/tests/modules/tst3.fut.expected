18373487b76f9acf656fa00b148e3431
